File: Note/app/src/main/java/com/zukron/note/model/DefaultNote.java
package com.zukron.note.model;

/**
 * Project name is Note
 * Created by Zukron Alviandy R on 8/7/2020
 */
public class DefaultNote {
    private Integer id;
    private String title;
    private String body;

    public DefaultNote(Integer id, String title, String body) {
        this.id = id;
        this.title = title;
        this.body = body;
    }

    public Integer getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getBody() {
        return body;
    }

}

File: Note/app/src/main/java/com/zukron/note/model/DefaultNote.kt
package com.zukron.note.model

import androidx.room.*
import androidx.room.ForeignKey.CASCADE

/**
 * Project name is Note
 * Created by Zukron Alviandy R on 8/7/2020
 */
@Entity(
        tableName = "default_note",
        foreignKeys = [ForeignKey(
                entity = Note::class,
                parentColumns = arrayOf("id"),
                childColumns = arrayOf("note_id"),
                onUpdate = CASCADE,
                onDelete = CASCADE
        )],
        indices = [Index(value = ["note_id"], unique = true)]
)
data class DefaultNote(
        var body: String
) {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var title: String = ""

    @ColumnInfo(name = "note_id")
    var noteId: Long = 0
}